overlord/servicestate/quota_control.go

```diff
@@ -83,6 +84,53 @@ func quotaGroupsAvailable(st *state.State) error {
 	return nil
 }
 
+func isQuotaAvailable(st *state.State, feature features.SnapdFeature, settingName string) error {
+	tr := config.NewTransaction(st)
+	status, err := features.Flag(tr, feature)
+	if err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	if !status {
+		return fmt.Errorf("experimental feature disabled - test it by setting 'experimental.%s' to true", settingName)
+	}
+	return nil
+}
+
+func verifyQuotaRequirements(st *state.State, resourceLimits quota.Resources) error {
+	// validate quotas in general are available
+	if err := quotaGroupsAvailable(st); err != nil {
+		return err
+	}
+
+	// Upon initialization verification for systemd version 230 has already been done,
+	// but for some of these quota types we need even higher version.
+	// see: EnsureQuotaUsability
+
+	// MemoryLimit requires systemd 211, so it's covered by the initial check
+	// CPUQuota requires systemd 213, so no further checks need to be done
+	// TasksMax requires systemd 228, so no further checks need to be done
+
+	// AllowedCPUs requires systemd 243, so we need to verify the version here
+	if resourceLimits.CPUSet != nil {
+		if err := systemd.EnsureAtLeast(243); err != nil {
+			return fmt.Errorf("cannot use the cpu-set quota with incompatible systemd: %v", err)
+		}
+	}
+
+	// Journal quotas require systemd 245, so we need to verify the version here as well
+	if resourceLimits.Journal != nil {
+		if err := systemd.EnsureAtLeast(245); err != nil {
+			return fmt.Errorf("cannot use journal quota with incompatible systemd: %v", err)
+		}
+
+		// We also need to check against experimental features for journal quota
+		if err := isQuotaAvailable(st, features.JournalQuota, "journal-quota"); err != nil {
```

Comment:
what happens on user systems, right now the set quota-groups and they get journal? but now they need to set journal-quota or they will not get journal quotas anymore?

Reply:
If any user is currently using Journal Quota, the quota groups with the journal quotas are still in effect and working. However if they try to create a new quota group with a journal quota or modify an existing group to add/change a journal quota, then they will be asked to turn on the feature.

You can always remove a group without any restrictions in this regard. So if you don't want to turn on the feature (I don't see why you wouldn't though), then you can always remove the group.

Reply:
The issue is that if it's a software agent doing those operations, things will be broken on a snapd update until that agent is changed to turn on the feature. At this point it would be more conservative to change the condition for journal quota to: quota-groups OR journal-quota. This also means for now not being able to just change the default of quota-groups but simply remove the other checks as you mentioned.

Reply:
Ah that makes sense. I think that would be a good solution to not breaking existing usage of journal quota. I'll change the check to quota-groups OR journal-quota.

Reply:
This is done!